lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -1102,6 +1114,30 @@ private boolean checkTablePermission(
         return false;
     }
 
+    private boolean checkFunctionPermission(SystemSecurityContext context, String functionName, Function<CatalogFunctionAccessControlRule, Boolean> executePredicate)
+    {
+        Identity identity = context.getIdentity();
+        return functionRules.stream()
+                .filter(functionRule -> functionRule.matches(identity.getUser(), identity.getEnabledRoles(), identity.getGroups(), functionName))
+                .map(executePredicate)
+                .findFirst()
+                .orElse(false);
+    }
+
+    private boolean checkFunctionPermission(SystemSecurityContext context, FunctionKind functionKind, CatalogSchemaRoutineName functionName, Function<CatalogFunctionAccessControlRule, Boolean> executePredicate)
+    {
+        AccessMode requiredCatalogAccess = switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW -> READ_ONLY;
+            case TABLE -> ALL;
+        };
+        Identity identity = context.getIdentity();
+        return canAccessCatalog(context, functionName.getCatalogName(), requiredCatalogAccess) && functionRules.stream()
+                .filter(functionRule -> functionRule.matches(identity.getUser(), identity.getEnabledRoles(), identity.getGroups(), functionKind, functionName))
+                .map(executePredicate)
+                .findFirst()
+                .orElse(false);
```

Comment:
misleading formatting

Reply:
I put `functionRules.stream()` in a next line